src/simulation/dynamics/constraintEffector/_UnitTest/test_InOutMessageFiltering.py

```diff
@@ -0,0 +1,369 @@
+# ISC License
+#
+# Copyright (c) 2024, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+#
+#   Unit Test Script
+#   Module Name:        constraintEffector
+#   Author:             Ananya Kodukula
+#   Creation Date:      July 19, 2024
+#
+
+import inspect
+import os
+import pytest
+import matplotlib.pyplot as plt
+import numpy as np
+
+from Basilisk.utilities import SimulationBaseClass, unitTestSupport, orbitalMotion, macros, RigidBodyKinematics
+from Basilisk.simulation import spacecraft, constraintDynamicEffector, gravityEffector, svIntegrators
+from Basilisk.architecture import messaging
+
+# uncomment this line if this test is to be skipped in the global unit test run, adjust message as needed
+# @pytest.mark.skipif(conditionstring)
+# uncomment this line if this test has an expected failure, adjust message as needed
+# @pytest.mark.xfail()
+
+@pytest.mark.parametrize("CutOffFreq,useConstEffector",[
+    (0.1,0), #Constraint Dynamic effector not connected
+    (0.1,1), #Constraint Dynamic effector connected
+    (0.1,-1), #Constraint Dynamic effector default state (connected)
+    (-1,1), # Negative Cut off frequency test
+    (0,1)]) #Zero cut off frequency test
+
+def test_constraintEffectorAllCases(show_plots,CutOffFreq,useConstEffector):
+
+def constraintEffectorInOutMessageFiltering(show_plots,wc,deviceStatus):
```

Comment:
I'm not following the code here, you have one empty `test_constraintEffectorAllCases` definition followed by the next function definition?  Maybe my python knowledge is too limited here.  Thanks for explaining

Reply:
Yes, agreed. The second function definition was redundant. It has been updated